packer/scripts/20-caddy.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+
+groupadd --system caddy
+useradd --system \
+    --gid caddy \
+    --create-home \
+    --home-dir /var/lib/caddy \
+    --shell /usr/sbin/nologin \
+    --comment "Caddy web server" \
+    caddy
+
+chown -R caddy:caddy /var/lib/caddy
+
+VERSION_SLUG=$(echo $CADDY_VERSION | sed 's/-/_/g')
+wget -qO /usr/bin/caddy https://github.com/caddyserver/caddy/releases/download/v${CADDY_VERSION}/caddy${VERSION_SLUG}_linux_amd64
```

Comment:
@mholt , can we standardize how the version and binary name will look like? The `sed` operation here turns `v2.0.0-beta9` to `v2.0.0_beta9`. That's not how you've named the binaries in releases of v2. The last one was `caddy2_beta9_linux_amd64`. To go from `v2.0.0-beta9` to `caddy2_beta9` I can do the following line:

`VERSION_SLUG=$(echo $CADDY_VERSION | sed 's/-/_/g' | sed 's/\.[0-9]*\.[0-9]*//g')`

But you'd lose the point releases in the binary name. Let's agree on a format to better standardize our scripts along with binary names.

Reply:
Sure. Right now I'm manually building and naming each binary, but once the process is automated (probably sometime around the release candidates), they'll be consistent.

Would it just be easier to use the version tag, unchanged, in the name?

Also, note that the automated builds will probably be zip/tar files archives.

Reply:
So something like this

```
ver=$(git describe --abbrev=0)
geese=("linux" "windows" "macos")
goarches=("amd64" "x86")

for goos in "${geese[@]}"
do
    for goarch in "${goarches[@]}"
    do
        GOOS=$goos GOARCH=$goarch go build -o caddy${ver}_${os}_${arch} -trimpath main.go
    done
done
```
?

We can adjust this file later when we start having automated zip/tar archives.